common/lib/dependabot/logger.rb

```diff
@@ -3,11 +3,26 @@
 require "logger"
 
 module Dependabot
-  def self.logger
-    @logger ||= Logger.new(nil)
-  end
+  module Logger
+    class Formatter < ::Logger::Formatter
+      def call(severity, _datetime, _progname, msg)
+        "#{severity} #{msg2str(msg)}\n"
+      end
+    end
+
+    def logger
+      @logger ||= ::Logger.new(Dependabot::Logger.destination).tap do |logger|
+        logger.progname = self.class.name.split("::").last
+        logger.formatter = Formatter.new
+      end
+    end
+
+    def self.destination
+      @destination ||= nil
+    end
 
-  def self.logger=(logger)
-    @logger = logger
+    def self.destination=(destination)
+      @destination = destination
```

Comment:
I know we're just moving this code around, but I'm just noticing now that we need to set the destination before calling the logger or it will noop, and I'm wondering if that might be confusing

Reply:
Ah yes good catch, previous design of setting a logger might have been more straight forward as we don't need the formatter here and stuff here